internal/services/workloads/workloads_sap_three_tier_virtual_instance_resource.go

```diff
@@ -0,0 +1,1961 @@
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+
+					"high_availability_type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringInSlice(sapvirtualinstances.PossibleValuesForSAPHighAvailabilityType(), false),
+					},
+
+					"secondary_ip_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						ForceNew: true,
+						Default:  false,
+					},
+
+					"transport_create_and_mount": {
+						Type:          pluginsdk.TypeList,
+						Optional:      true,
+						ForceNew:      true,
+						MaxItems:      1,
+						ConflictsWith: []string{"three_tier_configuration.0.transport_mount"},
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"resource_group_id": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: commonids.ValidateResourceGroupID,
+								},
+
+								"storage_account_name": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: storageValidate.StorageAccountName,
+								},
+							},
+						},
+					},
+
+					"transport_mount": {
+						Type:          pluginsdk.TypeList,
+						Optional:      true,
+						ForceNew:      true,
+						MaxItems:      1,
+						ConflictsWith: []string{"three_tier_configuration.0.transport_create_and_mount"},
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"file_share_id": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: storageValidate.StorageShareResourceManagerID,
+								},
+
+								"private_endpoint_id": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: privateendpoints.ValidatePrivateEndpointID,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"managed_resource_group_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: resourcegroups.ValidateName,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r WorkloadsSAPThreeTierVirtualInstanceResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r WorkloadsSAPThreeTierVirtualInstanceResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+
+			if v := rd.Get("three_tier_configuration.0.database_server_configuration.0.disk_volume_configuration"); v != nil {
+				diskVolumes := v.(*pluginsdk.Set).List()
+				if hasDuplicateVolumeName(diskVolumes) {
+					return fmt.Errorf("`volume_name` cannot be duplicated")
```

Comment:
shall we add a testcase for this error?

Reply:
I assume that we don't need to add a test case for a special logic based on below reasons.

Reason 1: I tested this logic on my local and all related scenarios are passed.
Reason 2: This is not a critical blocking issue. 
Reason 3: Generally, we don't need to add test case against the special logic since this kind of logic wouldn't be changed once it's added.